### PR TITLE
general: migrate Flask-SSO-SAML code

### PR DESCRIPTION
**Pull request by egabancho**

### CODE REVIEW

File: invenio_saml/ext.py [hunk 1]
```diff
@@ -1,21 +1,239 @@
...
+from collections.abc import Mapping
+from functools import wraps
 
-from flask_sso_saml import FlaskSSOSAML
+from flask import url_for
+from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser
+from werkzeug.utils import cached_property, import_string
 
-from . import config
+from invenio_saml import config
```
Comment (utnapischtim) [inline invenio_saml/ext.py:18]: Minor: is there a reason not to use the relative import for package specific imports?

File: invenio_saml/handlers.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def account_info(attributes, remote_app):
         external_id=external_id,
         external_method=remote_app,
         active=True,
-        confirmed_at=datetime.utcnow().isoformat()
+        confirmed_at=datetime.utcnow()
```
Comment (utnapischtim) [inline invenio_saml/handlers.py:69]: What is the reason of changing the confirmed_at date format?

File: invenio_saml/ext.py (hunk 1 above)
Comment (egabancho) [reply invenio_saml/ext.py:18]: That's just my editor taking over control, I'll revert the change ;-)

File: invenio_saml/handlers.py (hunk 2 above)
Comment (egabancho) [reply invenio_saml/handlers.py:69]: Good question; I forgot to write a comment myself.
When running the tests, I encounter an issue with this and SQLite, something to do with being unable to parse dates from strings.
But now that you point this out, SQLite is not part of the test matrix, so we can probably put the formatted output back.

Comment (utnapischtim) [reply invenio_saml/handlers.py:69]: https://github.com/inveniosoftware/invenio-accounts/pull/428#discussion_r1024914387

ok we go with `datetime.utcnow()`

File: invenio_saml/config.py [hunk 3]
```diff
@@ -1,13 +1,197 @@
...
+    receives two parameters, an instance of ``SAMLAuth`` and the next url that
+    the user will be redirected to. It gets called after
+    ``OneLogin_Saml2_Auth.process_slo``.
+    Typically returns the next URL too, although it is not mandatory.
 """
+
+
+# Default handlers
+
+SSO_SAML_DEFAUTL_SETTINGS_HANDLER = None
```
Comment (rekt-hard) [inline invenio_saml/config.py:164]: Typos. Should be `DEFAULT`

File: invenio_saml/ext.py [hunk 4]
```diff
@@ -1,21 +1,239 @@
...
+                cert = cf.read()
+            config["settings"]["sp"]["x509cert"] = cert
+
+        if config["sp_key_file"]:
+            with open(config["sp_key_file"], "r") as cf:
+                cert = cf.read()
+            config["settings"]["sp"]["privateKey"] = cert
+
+        # Import handlers is present
+        config["settings_handler"] = make_handler(
```
Comment (rekt-hard) [inline invenio_saml/ext.py:211]: Typos. Should be `DEFAULT`

File: invenio_saml/config.py [hunk 5]
```diff
@@ -1,13 +1,197 @@
 # -*- coding: utf-8 -*-
 #
-# Copyright (C) 2019 Esteban J. Garcia Gabancho.
+# Copyright (C) 2019, 2022 Esteban J. G. Gabancho.
 #
-# Invenio-SAML is free software; you can redistribute it and/or modify it
+# Flask-SSO-SAML is free software; you can redistribute it and/or modify it
```
Comment (rekt-hard) [inline invenio_saml/config.py:5]: Typo

**Review by rekt-hard — COMMENTED** [review]
General remark: Sometimes `invenio-saml` is used, other times `invenio-sso-saml`. Personally, I would keep `invenio-saml`. The package itself can stay the same and does not have to be renamed. This would allow us to keep many config variables the same and not having to propagate the changes to other modules.

Either way, it has to be consistent through the package.

**Review by mb-wali — APPROVED** [review]

**Pull request merged**